pilot/pkg/bootstrap/configcontroller.go

```diff
@@ -221,7 +221,9 @@ func (s *Server) initConfigSources(args *PilotArgs) (err error) {
 				return fmt.Errorf("failed to dial XDS %s %v", configSource.Address, err)
 			}
 			store := memory.MakeSkipValidation(collections.Pilot)
-			configController := memory.NewController(store)
+			configController := memory.NewControllerWithOptions(store, &memory.Config{
+				HasSynced: xdsMCP.HasSynced,
```

Comment:
why not directly pass `func HasSynced() bool `

Reply:
I consider the fact that there're several `new` functions already. Then I add a general one which can do the complex init things.